java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -211,6 +211,40 @@ public void close() {
     viewHandle = 0;
   }
 
+  @Override
+  public String toString() {
+    return "ColumnView{" +
+           "rows=" + rows +
+           ", type=" + type +
+           ", nullCount=" + nullCount +
+           '}';
+  }
+
+  @Override
+  public int hashCode() {
+    return type.hashCode() +
+           Long.hashCode(rows) +
+           Long.hashCode(nullCount) +
+           Long.hashCode(viewHandle);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o instanceof ColumnView) {
+      ColumnView other = (ColumnView) o;
+      if (this == other) {
+        return true;
+      } else if (!type.equals(other.type) || rows != other.rows || nullCount != other.nullCount) {
+        return false;
+      } else {
+        throw new IllegalStateException("Cannot compare the column views by `equals`. " +
+                "Need to iterate its rows and compare one by one.");
+      }
+    } else {
+      return false;
+    }
+  }
```

Comment:
I don't think these should be added as part of this PR.  hashCode will invite others to use these objects in hash maps/sets which is inappropriate, and equals should not throw.

Reply:
Thanks for reivew, I removed these.
And filed an issue (https://github.com/rapidsai/cudf/issues/8090) to track this.